README.md

```diff
@@ -190,11 +190,488 @@ autotools-snap --files ./components/**/*.meta.ts
+- **after**: called once after all other hooks have been called. This is where you'd do any necessary cleanup (deleting temporary directories, files, etc.)
+
+The interface for `IFileInfo` is as follows:
+
+```ts
+interface IFileInfo {
+  basename: string;
+  filepath: string;
+  data: string;
+  cssPath?: string;
+}
+```
+
+`filepath` should include the basename. For example, a file named "blue.ts", located in "users/JimBob/Documents/colors/", and whose exported component uses css classes from `colors.css` located in the same folder should have the following FileInfo:
+
+```ts
+{
+    basename: 'blue.ts',
+    filepath: 'users/JimBob/Documents/colors/blue.ts',
+    data: 'import {red} from \'./red\';\nlet violet...',
+    cssPath: 'users/JimBob/Documents/colors/colors.css'
+}
+
+```
+
+The interface for the hook arguments (`ISnapInfo`, `ICompSnapInfo`, etc.) is described in the following section.
+
+Hooks themselves are run in a specific order (Project -> Component -> Sim -> After), but hooks of each type are not run in a specific order. The array of hooks is looped over, and each hook is called in turn. It is important to write your hook code so that the order of execution does not affect the data returned.
+
+The project, component, and sim hooks allow you to provide custom behaviour to fit your specific project configuration. Ui-Autotools ships with an assortment of hooks that may fit your project.
+
+- `stylable-snap-plugin`: for building and snapping stylable projects.
+- `wix-style-react-snap-plugin`: for wix/wix-style-react.
+
+###### Project Hook
+
+The project hooks are run at the beginning of Snap execution, and are passed an object of type `ISnapInfo`.
+
+``` ts
+interface ISnapInfo {
+  Registry: IRegistry;
+  projectPath: string;
+  tempDirPath: string;
+  collectFiles: (files: IFileInfo[]) => void;
+}
+```
+
+`ISnapInfo` contains: `Registry`, which will contain all the project metadata at the point it is passed to each hook. `projectPath` describes the base path of the project in which snap is being run. `tempDirPath` describes the path to a temporary directory where snapshot files (HTML files) should be written to. `collectFiles` is a method used by hooks to append files to the list which will be passed to Applitools.
+
+Project hooks must return a Promise, but that Promise is of type `any`. Whatever is returned from a project hook will be stored in an object called `projectHookData`, which is then passed to each hook. This is useful in certain cases, for example, if you wanted to create a temporary directory in your project hook, store generated files there, and then destroy the directory in the after hook.
+
+###### Component Hook
+
+Component hooks are run once for every component in the project which has an associated metadata file. Component hooks are passed an `ICompSnapInfo` object:
+
+```ts
+interface ICompSnapInfo extends ISnapInfo {
+  componentMetadata: IComponentMetadata<any, any>;
+  projectHookData: any;
```

Comment:
given your comment that `projectHookData` currently only gets to After hooks - should this stay?